lms/djangoapps/teams/views.py

```diff
@@ -363,6 +363,18 @@ def post(self, request):
                 ugettext_noop('The supplied course_id {course_id} is not valid.'),
                 course_id=course_id
             )
+            return Response({
+                'field_errors': field_errors,
+            }, status=status.HTTP_400_BAD_REQUEST)
+
+        if CourseTeamMembership.user_in_team_for_course(request.user, course_key):
```

Comment:
@dianakhuang Uck, we shouldn't be doing this check if the user is a member of course staff, global staff, or "privileged" (discussion admin/moderator/ta).